### PR TITLE
Fix issue 287

### PR DESCRIPTION
**Pull request by medamineziraoui**
Fix those issues

https://github.com/saagie/gradle-saagie-dataops-plugin/issues/301
https://github.com/saagie/gradle-saagie-dataops-plugin/issues/298
https://github.com/saagie/gradle-saagie-dataops-plugin/issues/303
https://github.com/saagie/gradle-saagie-dataops-plugin/issues/302

### CODE REVIEW

File: src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy [hunk 1]
```diff
@@ -1489,7 +1488,8 @@ class SaagieClient {
 			}
 			def listVariables = getListOfVariablesFromRequest(variablesListRequest)
 			if (listVariables.size().equals(0)) {
-				return null
+				logger.warn("statement that No variables found")
```
Comment (ZouhairBear) [inline src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy:1491]: can you make a more clear message ?

File: src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy [hunk 2]
```diff
@@ -1545,18 +1545,21 @@ class SaagieClient {
 			} else if (parsedResultForVariableList.data?.projectEnvironmentVariables) {
 				listVariables = parsedResultForVariableList.data.projectEnvironmentVariables
 			}
-			return listVariables;
+			return listVariables ;
```
Comment (DecampsRenan) [inline src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy:1548]: please remove space before the semicolon

File: src/main/groovy/io/saagie/plugin/dataops/models/JobOverride.groovy [hunk 3]
```diff
@@ -5,7 +5,7 @@ package io.saagie.plugin.dataops.models
  * This provides a flexible option to set environment specific values.
  * The values thats currently supported are isScheduled, cronScheduling, alerting.
  */
-class JobOverride extends PropertyOverride{
+class JobOverride extends ArtifactPropertyOverride{
```
Comment (DecampsRenan) [inline src/main/groovy/io/saagie/plugin/dataops/models/JobOverride.groovy:8]: Is that normal that this class is empty ?

File: src/main/groovy/io/saagie/plugin/dataops/models/PipelineOverride.groovy [hunk 4]
```diff
@@ -1,4 +1,4 @@
 package io.saagie.plugin.dataops.models
 
-class PipelineOverride extends PropertyOverride{
+class PipelineOverride extends ArtifactPropertyOverride{
```
Comment (DecampsRenan) [inline src/main/groovy/io/saagie/plugin/dataops/models/PipelineOverride.groovy:3]: same remark as above

File: src/main/groovy/io/saagie/plugin/dataops/models/JobOverride.groovy (hunk 3 above)
Comment (medamineziraoui) [reply src/main/groovy/io/saagie/plugin/dataops/models/JobOverride.groovy:8]: Well we structered like this thinking that there will be contain in those classes but it didn't happen, i will remove them

File: src/main/groovy/io/saagie/plugin/dataops/models/PipelineOverride.groovy (hunk 4 above)
Comment (medamineziraoui) [reply src/main/groovy/io/saagie/plugin/dataops/models/PipelineOverride.groovy:3]: Same response "we structered like this thinking that there will be contain in those classes but it didn't happen, i will remove them"

**Pull request merged**